(/authentication/social-connections/microsoft) ff: update guide (#1607)
clerk/clerk-docs: docs/authentication/social-connections/microsoft.mdx

  ### Connect your Entra ID app and get your redirect URI

  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
  1. Select **Add connection** and select **For all users**.
  1. In the **Choose provider** dropdown, select **Microsoft**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on. Then:
     - Under **Client ID**, add the value you copied from **Application (client) ID** in the Microsoft Azure portal.
     - Under **Client Secret**, add the client secret value you generated.
     - Copy the **Authorized redirect URI**. You'll need it for the final step to configure your Entra ID app.
     - Select **Add connection**.

  ### Enable OpenID

  To connect your Clerk app to your Microsoft app, set the **Authorized redirect URI** in your Microsoft Azure portal.

  1. Navigate back to the Microsoft Azure portal.
  1. In the sidebar, open the **Manage** dropdown and select **Authentication**.
  1. Select **Add a platform**.
  1. Select **Web**.

>   1. In the **Redirect URIs** field and the **Front-channel logout URL** field, add the **Authorized redirect URI** you copied earlier.

Reviewer: ```suggestion
  1. In the **Redirect URIs** field and the **Front-channel logout URL** field, add the **Authorized redirect URI** you copied from the Clerk Dashboard.
```

  1. Copy your new client secret's **Value** and save it securely. You'll add this and your client ID to your Clerk app later. Keep this page open.

  ### Connect your Entra ID app and get your redirect URI

  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
  1. Select **Add connection** and select **For all users**.
  1. In the **Choose provider** dropdown, select **Microsoft**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on. Then:
     - Under **Client ID**, add the value you copied from **Application (client) ID** in the Microsoft Azure portal.
     - Under **Client Secret**, add the client secret value you generated.
     - Copy the **Authorized redirect URI**. You'll need it for the final step to configure your Entra ID app.
     - Select **Add connection**.

  ### Enable OpenID

  To connect your Clerk app to your Microsoft app, set the **Authorized redirect URI** in your Microsoft Azure portal.

  1. Navigate back to the Microsoft Azure portal.

>   1. In the sidebar, under **Manage**, select **Authentication**.

Reviewer: ```suggestion
  1. In the sidebar, open the **Manage** dropdown and select **Authentication**.
```

  - Protect your application from [the nOAuth exploit](https://www.descope.com/blog/post/noauth).
</TutorialHero>

Enabling OAuth with **Microsoft Azure Entra ID** (formerly [Active Directory](https://learn.microsoft.com/en-us/entra/fundamentals/new-name)) allows your users to sign in and sign up to your Clerk application with their Microsoft account.

## Configure for your development instance

For _development instances_, Clerk uses preconfigured shared OAuth credentials and redirect URIs—no other configuration is needed.

1. Navigate to the [**Clerk Dashboard**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections).
1. In the top navigation, select **Configure**. Then in the sidebar, select **SSO Connections**.
1. Select the **Add connection** button, and select **For all users**.
1. In the **Choose provider** dropdown, search for and select **Microsoft**.
1. Select **Add connection**.

## Configure for your production instance

In _production instances_, you must provide custom credentials, which involves generating your own **Client ID** and **Client Secret** using your Microsoft Entra ID account.

> To make the setup process easier, it's recommended to keep two browser tabs open: one for your [**Clerk Dashboard**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [**Microsoft Azure portal**](https://portal.azure.com).

Reviewer: ```suggestion
To make the setup process easier, it's recommended to keep two browser tabs open: one for your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [Microsoft Azure portal](https://portal.azure.com).
```

<Steps>
  ### Create a Microsoft Entra ID app

  > [!TIP]

>   > If you already have a Microsoft Entra ID app you'd like to connect to Clerk, select your app from the [**Microsoft Azure portal**](https://portal.azure.com/#home) homepage and skip to [the next step in this tutorial](#get-your-client-id-and-client-secret).

Reviewer: ```suggestion
  > If you already have a Microsoft Entra ID app you'd like to connect to Clerk, select your app from the [Microsoft Azure portal](https://portal.azure.com/#home) homepage and skip to [the next step in this tutorial](#get-your-client-id-and-client-secret).
```

>   1. On the homepage of the [**Microsoft Azure portal**](https://portal.azure.com/#home), in the **Azure services** section, select [**Microsoft Entra ID**](https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/Overview).

Reviewer: ```suggestion
  1. On the homepage of the [Microsoft Azure portal](https://portal.azure.com/#home), in the **Azure services** section, select [**Microsoft Entra ID**](https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/Overview).
```

  1. In the sidebar, open the **Manage** dropdown and select **[App registrations](https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps)**.
  1. Select **New Registration**. You'll be taken to the **Register an application** page.
  1. Fill out the form as follows:
     1. Under **Name**, name the app whatever you'd like. For example, "Clerk Demo App."
     1. Under **Supported account types**, select **Accounts in any organizational directory (Any Microsoft Entra ID tenant – Multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)**.
     1. Under **Redirect URI (Optional)**, select **Web**.
     1. Finally, select **Register** to submit the form.

  ### Get your client ID and client secret

>   Once your Microsoft Entra ID app is created, or once you select your app from the **Microsoft Azure portal**, you'll be taken to its **Overview** page.

Reviewer: ```suggestion
  Once your Microsoft Entra ID app is created, or once you select your app from the Microsoft Azure portal, you'll be taken to its **Overview** page.
```

  1. From your app's overview, copy the **Application (client) ID** and save it somewhere secure. It's required for connecting your Microsoft Entra ID app to your Clerk app.
  1. On this same page, under **Client credentials**, select **Add a certificate or secret** to generate a Client Secret. You'll be redirected to the **Certificate & secrets** page.
  1. Select the **New client secret** button. In the panel that opens, add a description and set an expiration time for your secret. Select **Add**.
     > [!IMPORTANT]
     > When your secret expires, your social connection will stop working until you generate a new client secret and add it to your Clerk app.
  1. Copy your new client secret's **Value** and save it somewhere secure. You'll add it to your Clerk application later, alongside your client ID.
  1. Leave this tab open.

>   ### Connect your Entra ID app and get your redirect URI

Reviewer: ```suggestion
  ### Connect your Microsoft Entra ID app and get your redirect URI
```

>   1. Navigate to the [**Clerk Dashboard**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections).

Reviewer: ```suggestion
  1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections).
```

  1. In the top navigation, select **Configure**. Then in the sidebar, select **SSO Connections**.
  1. Select the **Add connection** button, and select **For all users**.
  1. In the **Choose provider** dropdown, search for and select **Microsoft**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on. Then:
     - Under **Client ID**, paste the value you copied from **Application (client) ID** the Microsoft Azure portal.

>      - Under **Client Secret**, paste the client secret value you generated from Microsoft Entra ID.

Reviewer: should we keep this consistent?

```suggestion
     - Under **Client Secret**, paste the client secret value you generated from the Microsoft Azure portal.
```

>      - Copy the **Authorized redirect URI**. You need it for the final step to configure your Entra ID app.

Reviewer: ```suggestion
     - Copy the **Authorized redirect URI**. You need it for the final step to configure your Microsoft Entra ID app.
```

Reviewer: do we want to repeat Microsoft Entra ID app all throughout, or just Microsoft app? we need to settle on and use that

     - Select **Add connection**.

  ### Enable OpenID

>   To connect your Clerk app to your Microsoft app, you must set the **Authorized redirect URI** in your Microsoft Entra Azure portal.

Reviewer: I think we were using Microsoft Azure portal and not Microsoft Entra Azure portal

```suggestion
  To connect your Clerk app to your Microsoft app, you must set the **Authorized redirect URI** in your Microsoft Azure portal.
```

>   1. Navigate back to the **Microsoft Azure portal**.

Reviewer: ```suggestion
  1. Navigate back to the Microsoft Azure portal.
```

  1. In the sidebar, open the **Manage** dropdown and select **Authentication**.
  1. Select **Add a platform**.
  1. Select **Web**.
  1. In the **Redirect URIs** field and the **Front-channel logout URL** field, paste the **Authorized redirect URI** you copied in the previous step.
  1. Under **Implicit grant and hybrid flows**, select both **Access tokens** and **ID tokens**.
  1. Select **Configure** to save the changes.

  ### Test your OAuth

  The simplest way to test your OAuth is to visit your Clerk application's [Account Portal](/docs/customization/account-portal/overview), which is available for all Clerk applications out-of-the-box.

>   1. In the navigation sidebar of the **Clerk Dashboard**, select [**Account Portal**](https://dashboard.clerk.com/last-active?path=account-portal).

Reviewer: ```suggestion
  1. In the navigation sidebar of the Clerk Dashboard, select [**Account Portal**](https://dashboard.clerk.com/last-active?path=account-portal).
```

  1. Next to the **Sign-in** URL, select **Visit**. The URL should resemble:
     - **For development** – `https://your-domain.accounts.dev/sign-in`
     - **For production** – `https://accounts.your-domain.com/sign-in`
  1. On the sign-in page, you should see **Microsoft** as an option. Select it to sign in with your Microsoft account.

  ### Secure your app against the nOAuth vulnerability

  [nOAuth](https://www.descope.com/blog/post/noauth) is an exploit in Microsoft Entra ID OAuth applications that can lead to full account takeovers via email address spoofing. To protect users, Clerk enforces stricter checks on verified email addresses.

  For further security, Microsoft offers an optional [**`xms_edov`**](https://learn.microsoft.com/en-us/entra/identity-platform/migrate-off-email-claim-authorization#using-the-xms_edov-optional-claim-to-determine-email-verification-status-and-migrate-users) claim, which provides additional context that can be used to determine whether the returned email is verified.

  To enable this optional claim, follow these steps:

>   1. Navigate to your Azure application in the **Microsoft Azure portal**.

Reviewer: ```suggestion
  1. Navigate to your Azure application in the Microsoft Azure portal.
```
